src/layers/schematicInterfaces.ts

```diff
@@ -105,6 +106,124 @@ export interface Cement {
   toc: number;
 }
 
+export type PerforationSubKind =
+  | 'Perforation'
+  | 'Open hole gravel pack'
+  | 'Open hole screen'
+  | 'Open hole'
+  | 'Open hole frac pack'
+  | 'Cased hole frac pack'
+  | 'Cased hole gravel pack'
+  | 'Cased hole fracturation';
+
+export interface Perforation {
+  kind: 'perforation';
+  subKind: PerforationSubKind;
+  id: string;
+  top: number;
+  bottom: number;
+  isOpen: boolean;
+  casingIds: string[];
+  holeId?: string[];
+}
+
+export type PerforationShape = ComplexRopeSegment;
```

Comment:
Think this can be moved to [`schematicShapeGenerator`](https://github.com/equinor/esv-intersection/blob/master/src/datautils/schematicShapeGenerator.ts) 🤔